### ISSUE TITLE
networkd does not support DNS List option in RA

### ISSUE THREAD

**Issue by cvmiller** [systemd/systemd#4590]
### Submission type

  - [ ] Bug report
  - [x] Request for enhancement (RFE)

*NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!*

### systemd version the issue has been seen with

> systemd 231
+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN

*NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!*

### Used distribution

> Arch Linux

### In case of bug report: Expected behaviour you didn't see

> Search domains not placed in /etc/resolv.conf, as per RFC 6106

### In case of bug report: Unexpected behaviour you saw

> No search domains placed in /etc/resolv.conf

### In case of bug report: Steps to reproduce the problem

1. Setup Router with M-bit disabled (off) and RDNSS & DNSSL configured (a SLAAC Only config)
2. Boot machine
3. ```cat /etc/resolv.conf```, note **no** DNS search lines in file

**Comment by poettering** [comment]
we support this just fine. Do you have resolved enabled properly? Does "systemd-resolve --status" show the domains?

**Comment by cvmiller** [comment]
When I look at the domain I am sending in the RA, I don't see it on 'system-resolve --status'

```
$ rdisc6 eth0
Soliciting ff02::2 (ff02::2) on eth0...

Hop limit                 :           64 (      0x40)
Stateful address conf.    :          Yes
Stateful other conf.      :          Yes
Mobile home agent         :           No
Router preference         :       medium
Neighbor discovery proxy  :           No
...
 Recursive DNS server     : 2001:470:ebbd:0:211:24ff:fee1:dbc8
 Recursive DNS server     : 2001:470:ebbd::1
  DNS servers lifetime    :         1200 (0x000004b0) seconds
 DNS search list          : hoomaha.net   <==========================
  DNS search list lifetime:         1200 (0x000004b0) seconds

$ systemd-resolve --status
Global
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test
Link 4 (wlan0)
      Current Scopes: none
       LLMNR setting: yes
...
```

**Comment by poettering** [comment]
You truncated the most relevant part of "systemd-resolve --status": the bit about your eth0 link where this data should have been collected...

**Comment by cvmiller** [comment]
I am on the road for a couple of days. Let me try again, when I return.

**Comment by cvmiller** [comment]
Here's the entire output of "systemd-resolve --status" Note my DNS List (hoomaha.net) is not listed.
```
Global
          DNSSEC NTA: 10.in-addr.arpa
                      16.172.in-addr.arpa
                      168.192.in-addr.arpa
                      17.172.in-addr.arpa
                      18.172.in-addr.arpa
                      19.172.in-addr.arpa
                      20.172.in-addr.arpa
                      21.172.in-addr.arpa
                      22.172.in-addr.arpa
                      23.172.in-addr.arpa
                      24.172.in-addr.arpa
                      25.172.in-addr.arpa
                      26.172.in-addr.arpa
                      27.172.in-addr.arpa
                      28.172.in-addr.arpa
                      29.172.in-addr.arpa
                      30.172.in-addr.arpa
                      31.172.in-addr.arpa
                      corp
                      home
                      internal
                      intranet
                      lan
                      local
                      private
                      test

Link 4 (wlan0)
      Current Scopes: none
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no

Link 3 (eth1)
      Current Scopes: none
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no

Link 2 (eth0)
      Current Scopes: DNS LLMNR/IPv4 LLMNR/IPv6
       LLMNR setting: yes
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 192.168.11.1
                      2001:470:ebbd:0:211:24ff:fee1:dbc8
                      2001:470:ebbd::1
[root@alarm alarm]#
```

**Comment by cvmiller** [comment]
I have provided full output (see above). Can we remove the "needs-reporter-feedback' label?

**Comment by skimj** [comment]
Could this be related to the search domains option issue I submitted here: #8174 ? I don't have any experience with IPv6, RA, etc. I know they are totally different but could the systemd handling be related?

**Comment by cvmiller** [comment]
No, I am afraid it doesn't. IPv6 has two ways of getting an address, DNS info, and DNS search list, 1) via DHCPv6 (issue #8174 ) and via the Router Advertisement (aka Stateless Address Auto Config, or SLAAC).

Android and ChromeOS _only_ support SLAAC, which is why I have opened this issue.

**Comment by keszybz** [comment]
@yuwata what's the status here, are we still missing something?

**Comment by yuwata** [comment]
We have supported DNSSL provided through RA for a long time. By 1e7a0e21c97ac1bbc743009e5ec8c12bc6200e19. Hmm, but the original report is after the commit. @cvmiller Do you still have the issue?

**Comment by yuwata** [comment]
I've added a test for DNSSL in #17440. And it seems to work fine.
